lib/Minz/Log.php

```diff
@@ -70,12 +71,32 @@ public static function record ($information, $level, $file_name = null) {
 			     . ' [' . $level_label . ']'
 			     . ' --- ' . $information . "\n";
 
+			self::checkLogfileSize($file_name);
+
 			if (file_put_contents($file_name, $log, FILE_APPEND | LOCK_EX) === false) {
 				throw new Minz_PermissionDeniedException($file_name, Minz_Exception::ERROR);
 			}
 		}
 	}
 
+	/**
+	 * Make sure we do not waste a huge amount of disk space with old log messages.
+	 *
+	 * This method can be called multiple times for one script execution, but its result will not change unless
+	 * you call clearstatcache() in between. We won't due do that for performance reasons.
+	 *
+	 * @param $file_name
+	 * @throws Minz_PermissionDeniedException
+	 */
+	protected static function checkLogfileSize($file_name) {
+		$maxSize = defined('MAX_LOG_SIZE') ? MAX_LOG_SIZE : 512000;
+		if (@filesize($file_name) > $maxSize) {
+			if (file_put_contents($file_name, '') === false) {
```

Comment:
I would like to keep the newest messages.
I think the algorithm could be to divide the logfile by 2 when it increases above a threshold (delete first half of the file).
It is probably not needed to care about keeping entire lines.

Reply:
Sure, go ahead and adapt the PR. I don't see a use-case that justifies the effort. After using FreshRSS for more than a year, I did not take even one look into the log - its running to smooth ;-) I did not even know that it existed until I read #1562 and found out that my log file was impressive large as well.

Reply:
I will :-)

Reply:
Done in https://github.com/FreshRSS/FreshRSS/pull/1712/commits/3a4f23ba232ee1b643097092fd14a2b3380714ab